ambari-server/src/main/java/org/apache/ambari/server/controller/AmbariManagementControllerImpl.java

```diff
@@ -1396,6 +1398,16 @@ private Stage createNewStage(long id, Cluster cluster, long requestId,
       }
     }
 
+    if (request.getComponentId() != null) {
+      Service service = cluster.getServiceByComponentId(request.getComponentId());
```

Comment:
Hi Attila, the componentId in the request will be the one from the hostComponentState table but the getServiceByComponentId looks for the id from the serviceComponentDesiredState table. So although your GET /hostcomponents will return results, they are most likely pointing to incorrect ids so when you do GET /hostcomponents/{id} it will return resource not found.
cc @swapanshridhar

Reply:
We can do something like this
Service service = cluster.getServicesById().get(hostComponentStateDAO.findById(request.getComponentId()).getServiceId());